packages/dds/tree/src/test/simple-tree/treeApi.spec.ts

```diff
@@ -768,5 +768,147 @@ describe("treeNodeApi", () => {
 			assert.equal(nodeChanged, true, "'nodeChanged' should have fired");
 			assert.equal(treeChanged, true, "'treeChanged' should have fired");
 		});
+
+		it(`'nodeChanged' includes the names of changed properties (objectNode)`, () => {
+			const sb = new SchemaFactory("test");
+			class treeSchema extends sb.object("root", {
+				prop1: sb.optional(sb.number),
+				prop2: sb.optional(sb.number),
+				prop3: sb.optional(sb.number),
+			}) {}
+
+			const view = getView(new TreeViewConfiguration({ schema: treeSchema }));
+			view.initialize({ prop1: 1, prop2: 2 /* , prop3: 0 */ });
+			const root = view.root;
+
+			let listenerFired = false;
+			Tree.on(root, "nodeChanged", ({ changedProperties }) => {
+				// This is the main validation for the test
+				assert.deepEqual(changedProperties, new Set(["prop1", "prop2", "prop3"]));
+				listenerFired = true;
+			});
+
+			const { forkView, forkCheckout } = getViewForForkedBranch(view);
+
+			// The implementation details of the kinds of changes that can happen inside the tree are not exposed at this layer.
+			// But since we know then, try to cover all of them.
```

Comment:
```suggestion
			// But since we know them, try to cover all of them.
```
or

```suggestion
			// But since we know, then try to cover all of them.
```

Reply:
Thanks, good catch